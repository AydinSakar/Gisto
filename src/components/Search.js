import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { size, isEmpty } from 'lodash/fp';
import styled from 'styled-components';
import * as snippetActions from 'actions/snippets';
import { SIDEBAR_WIDTH, MINIMUM_CHARACTERS_TO_TRIGGER_SEARCH } from 'constants/config';
import { filterSnippetsList, isTag } from 'utils/snippets';
import Button from 'components/common/Button';
import Icon from 'components/common/Icon';
import { baseAppColor, borderColor } from 'constants/colors';
import Input from 'components/common/Input';

const SearchWrapper = styled.div`
  position: relative;
  padding: 0 10px;
  flex-direction: row;
  display: flex;
  width: ${SIDEBAR_WIDTH - 20}px;
  min-width: ${SIDEBAR_WIDTH - 20}px;
  color: #555;
  border-right: 1px solid ${borderColor};
  align-items: center;
  }
`;

const SearchTypeLabel = styled.span`
  position: absolute;
  font-size: 10px;
  top: 7px;
  left: 47px;
  height: auto;
  line-height: 1;
`;

const Search = ({ snippets, filterSnippets, filterText }) => {
  const countSnippets = size(filterSnippetsList(snippets, filterText));
  const searchType = isTag(filterText) ? 'tag' : 'free text';
  const shouldFilter = (query) => {
    return size(query) > MINIMUM_CHARACTERS_TO_TRIGGER_SEARCH || isEmpty(query);
  };

  return (
    <SearchWrapper>
      <Icon type="search" size="46" color={ baseAppColor }/>

      { filterText !== '' && (
        <SearchTypeLabel>
          Search by <strong>{ searchType }</strong> ({ countSnippets })
        </SearchTypeLabel>
      ) }

      <Input type="search"
             placeholder={ `Search ${countSnippets} snippets` }
             onChange={
               (event) => shouldFilter(event.target.value) && filterSnippets(event.target.value)
             }/>

      <Button icon="add">New snippet</Button>

    </SearchWrapper>
  );
};

Search.propTypes = {
  snippets: PropTypes.object,
  filterSnippets: PropTypes.func,
  filterText: PropTypes.string
};

const mapStateToProps = (state) => ({
  snippets: state.snippets.snippets,
  filterText: state.snippets.filter.text
});

export default connect(mapStateToProps, {
  filterSnippets: snippetActions.filterSnippets
})(Search);
