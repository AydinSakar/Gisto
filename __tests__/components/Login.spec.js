import { LogIn } from 'components/LogIn';

const propSetup = (props) => ({
  ...props
});

const setup = (props) => shallow(<LogIn { ...propSetup(props) }/>);

describe('COMPONENTS - <LogIn>', () => {
  test('render LogIn', () => {
    const component = setup({
      loginBasic: jest.fn(),
      loginWithToken: jest.fn(),
      twoFactorAuth: '',
      loading: false
    });

    expect(component).toMatchSnapshot();
  });
});
