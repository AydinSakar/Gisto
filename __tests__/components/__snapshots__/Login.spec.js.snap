// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENTS - <LogIn> render LogIn 1`] = `
<LoginWrapper>
  <Logo>
    { Gisto }
  </Logo>
  <small>
    v
    1.9.32
  </small>
  <div>
    <h4>
      Sign-in using 
      Github
       username and password
    </h4>
    <Input
      onChange={[Function]}
      placeholder="Email or username"
      type="text"
    />
    <br />
    <Input
      onChange={[Function]}
      placeholder="Password"
      type="password"
    />
    <br />
    <ResetLoginType>
      <Anchor
        onClick={[Function]}
      >
        Log-in with GitHub
      </Anchor>
      <Anchor
        onClick={[Function]}
      >
        Cancel
      </Anchor>
    </ResetLoginType>
    <br />
    <br />
    <Button
      icon="success"
      onClick={[Function]}
    >
      Log-in
    </Button>
    <br />
  </div>
  <p
    className="options"
  >
    <strong>
      Other options:
    </strong>
      sign-in using GitHub  
    <Anchor
      onClick={[Function]}
    >
      token
    </Anchor>
     | 
    <Anchor
      onClick={[Function]}
    >
      username and password
    </Anchor>
     | 
    <Anchor
      onClick={[Function]}
    >
      Use enterprise
    </Anchor>
  </p>
</LoginWrapper>
`;
